terminate/preview_quick.go

```diff
@@ -59,235 +44,45 @@ func PreviewTerminateSectorsQuick(
-					sectors, err := partition.LiveSectors.All(math.MaxUint64)
-					if err != nil {
-						errorCh <- err
-						return
-					}
-					sampledSectors := make([]uint64, 0)
-					partitionBatchSize := max(batchSize, desiredMinSamples/uint64(len(deadlinePartitions)))
-					step := max(float64(len(sectors))/float64(partitionBatchSize-1), 1.0)
-					lastIndex := -1
-					for sampleIndex := 0.0; int(sampleIndex) < len(sectors); sampleIndex += step {
-						i := int(sampleIndex)
-						if i > lastIndex {
-							sampledSectors = append(sampledSectors, sectors[i])
-							lastIndex = i
-						}
-					}
-					if lastIndex < len(sectors)-1 {
-						sampledSectors = append(sampledSectors, sectors[len(sectors)-1])
-					}
-					sampledSectorsBitfield := bitfield.NewFromSet(sampledSectors)
-					termination := miner.TerminationDeclaration{
-						Deadline:  uint64(dlIdx),
-						Partition: uint64(partIdx),
-						Sectors:   sampledSectorsBitfield,
-					}
-					terminationsCh <- terminationTask{
-						termination:         termination,
-						deadlineHeight:      deadlineHeight,
-						deadlineTs:          deadlineTs,
-						sectorsCount:        int64(sc),
-						sampledSectorsCount: int64(len(sampledSectors)),
-					}
-					sectorsTerminated += uint64(len(sampledSectors))
-					sectorsCount += uint64(len(sectors))
-				} else {
-					// DeadlinePartition not sampled
-					sectorsInSkippedPartitions += sc
-					sectorsCount += sc
-				}
-			}
-		}
-		close(terminationsCh)
-	}()
-
-	allStats := NewSectorStats()
-
-loop:
-	for {
-		select {
-		case stats := <-statsCh:
-			if stats == nil {
-				break loop
-			}
-			allStats = allStats.Accumulate(stats)
-		case err := <-errorCh:
-			return nil, err
-		}
-	}
 
-	if sectorsCount > 0 {
-		allStats = allStats.ScaleUp(int64(sectorsCount),
-			int64(sectorsCount-sectorsInSkippedPartitions))
+	stats := SectorStats{
+		TerminationPenalty: res.EstimatedTotalTerminationFee,
+		InitialPledge:      res.InitialPledge,
+		SectorCount:        res.SectorCount,
 	}
 
 	return &PreviewTerminateSectorsReturn{
 		Actor:                      actor,
 		MinerInfo:                  minerInfo,
 		VestingBalance:             lf.VestingFunds.Int,
 		InitialPledge:              new(big.Int).Add(lf.InitialPledgeRequirement.Int, lf.PreCommitDeposits.Int),
-		SectorStats:                allStats,
-		SectorsTerminated:          sectorsTerminated,
-		SectorsCount:               sectorsCount,
-		SectorsInSkippedPartitions: sectorsInSkippedPartitions,
-		Epoch:                      h,
-		PartitionsCount:            uint64(len(deadlinePartitions)),
-		SampledPartitionsCount:     sampledDeadlinePartitionCount,
+		SectorStats:                &stats,
+		SectorsTerminated:          uint64(res.SectorCount),
+		SectorsCount:               uint64(res.SectorCount),
+		SectorsInSkippedPartitions: 0,
+		Epoch:                      ts.Height(),
+		PartitionsCount:            64,
+		SampledPartitionsCount:     64,
```

Comment:
I'd probably just set these to 0, i dont think we use them and i dont think this is correct